quic/api/QuicTransportBase.cpp

```diff
@@ -1636,10 +1636,19 @@ QuicTransportBase::notifyPendingWriteOnStream(StreamId id, WriteCallback* wcb) {
   if (!qStream->writable()) {
     return folly::makeUnexpected(LocalErrorCode::STREAM_CLOSED);
   }
+
+  if (wcb == nullptr) {
+    return folly::makeUnexpected(LocalErrorCode::INVALID_WRITE_CALLBACK);
+  }
   // Add the callback to the pending write callbacks so that if we are closed
   // while we are scheduled in the loop, the close will error out the callbacks.
-  if (!pendingWriteCallbacks_.emplace(id, wcb).second) {
-    return folly::makeUnexpected(LocalErrorCode::INVALID_WRITE_CALLBACK);
+  auto existing_wcb_pair = pendingWriteCallbacks_.emplace(id, wcb);
+  if (existing_wcb_pair.second) {
+    if ((existing_wcb_pair.first)->second != wcb) {
+      return folly::makeUnexpected(LocalErrorCode::INVALID_WRITE_CALLBACK);
+    } else {
```

Comment:
this else case won't happen.

Reply:
sorry it's the if case that won't happen.

Reply:
with `!existing_wcb_pair.second` on L1646, the if case on L1647 can happen, correct?

Reply:
ye with the new update it can work